src/main/java/org/snomed/snowstorm/core/data/services/servicehook/CommitServiceHookClient.java

```diff
@@ -51,35 +65,52 @@ public void preCommitCompletion(Commit commit) throws IllegalStateException {
 			return;
 		}
 
-		final String authenticationToken = SecurityUtil.getAuthenticationToken();
 		try {
-			final HttpHeaders httpHeaders = new HttpHeaders();
-			httpHeaders.setContentType(MediaType.APPLICATION_JSON);
-			if (authenticationToken == null) {
-				logger.info("Authentication token is null.");
-			}
-			httpHeaders.add(HttpHeaders.COOKIE, authenticationToken);
-			restTemplate.postForEntity("/integration/snowstorm/commit",
+			String authenticationToken = SecurityUtil.getAuthenticationToken();
+			HttpHeaders httpHeaders = buildHttpHeaders(authenticationToken);
+			logRequest(commit, authenticationToken, commit.getBranch());
+			ResponseEntity<?> responseEntity = restTemplate.postForEntity("/integration/snowstorm/commit",
 					new HttpEntity<>(new CommitInformation(commit), httpHeaders), Void.class);
-		} catch (RestClientException e) {
-			logger.error("Commit service hook failed for branch {}, commit {}, url {}, cookie {}",
-					commit.getBranch().getPath(), commit.getTimepoint().getTime(), serviceUrl,
-					obfuscateToken(authenticationToken), e);
-			boolean promotion = commit.getCommitType().equals(Commit.CommitType.PROMOTION);
-			if (promotion && e instanceof HttpClientErrorException.Conflict) {
-				if (blockPromotion) {
-					logger.error("Promotion blocked; not all criteria have been met.");
-					throw new RuntimeServiceException("Promotion blocked; not all criteria have been met.");
-				}
+			logger.info("External system returned HTTP status code {}.", responseEntity.getStatusCodeValue());
+		} catch (ResourceAccessException e) {
+			// Cannot communicate with external system; perhaps lacking authentication or url configured incorrectly.
+			if (this.failIfError) {
+				logger.error("Cannot communicate with external system; throwing exception.");
+				throw new RuntimeServiceException("Cannot communicate with external system.", e);
+			} else {
+				logger.error("Cannot communicate with external system, however, failure will be ignored as per configuration.");
 			}
-
-			if (blockPromotion) {
-				logger.error("Promotion blocked; unexpected error.");
-				throw e;
+		} catch (HttpClientErrorException.Conflict e) {
```

Comment:
My understanding is that catch blocks are considered in top down order. I would be tempted to put this one as the first catch block because it's the most specific exception. The other catch which is less specific could continue to catch all instances of `RestClientException`, as long as it's below `HttpClientErrorException.Conflict`.

Reply:
Thanks Kai. 

I have made that change.